### PR TITLE
feat: [CDS-51558]: support for graph QL query for updating cf cli version  for services

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
This PR is for Supporting Graph Ql query for updating Cf Cli version in Services
Tests Done
* Tested the cf cli update for multiple application.
* Tested the update with giving exclude services.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43467)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/graphql/datafetcher/service/UpdateServiceMetadataDataFetcher.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    List<QLService> updatedServices = new ArrayList<>();
+    List<String> excludeServiceIds = new ArrayList<>();
+    List<QLApplication> applicationList = new ArrayList<>();
+
+    if (parameter.getExcludeServices().isPresent()) {
+      excludeServiceIds = parameter.getExcludeServices().getValue().orElse(Collections.emptyList());
+    }
+    final Application applicationForFFCheck = appService.get(parameter.getApplicationId().get(0));
+
+    if (!featureFlagService.isEnabled(FeatureName.CF_CLI7, applicationForFFCheck.getAccountId())) {
```
Comment (RishabhGupta34) [inline 400-rest/src/main/java/software/wings/graphql/datafetcher/service/UpdateServiceMetadataDataFetcher.java:81]: nit, this should be checked, if we are updating service to CLI7 right?

File: 400-rest/src/main/java/software/wings/graphql/schema/mutation/service/input/QLUpdateServiceMetadataInput.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import software.wings.graphql.schema.mutation.QLMutationInput;
+
+import java.util.List;
+import lombok.Builder;
+import lombok.Value;
+import lombok.experimental.FieldNameConstants;
+
+@Value
+@Builder
+@FieldNameConstants(innerTypeName = "QLUpdateApplicationInputKeys")
```
Comment (RishabhGupta34) [inline 400-rest/src/main/java/software/wings/graphql/schema/mutation/service/input/QLUpdateServiceMetadataInput.java:24]: nit, shouldn't the inner type name be diff.?

File: 400-rest/src/main/java/software/wings/graphql/datafetcher/service/UpdateServiceMetadataDataFetcher.java (hunk 1 above)
Comment (sourabh-awashti) [reply 400-rest/src/main/java/software/wings/graphql/datafetcher/service/UpdateServiceMetadataDataFetcher.java:81]: if the user is changing cf cli version then they must have this FF enabled, thats why this check

File: 400-rest/src/main/java/software/wings/graphql/schema/mutation/service/input/QLUpdateServiceMetadataInput.java (hunk 2 above)
Comment (sourabh-awashti) [reply 400-rest/src/main/java/software/wings/graphql/schema/mutation/service/input/QLUpdateServiceMetadataInput.java]: Done

File: 400-rest/src/main/java/software/wings/graphql/datafetcher/service/UpdateServiceMetadataDataFetcher.java (hunk 1 above)
Comment (sourabh-awashti) [reply 400-rest/src/main/java/software/wings/graphql/datafetcher/service/UpdateServiceMetadataDataFetcher.java:81]: even if they are changing from cf cli7 to cf cli 6 this is required

**Review by RishabhGupta34 — DISMISSED** [review]
> Reviewed 6 of 7 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43467)* status: 7 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by RishabhGupta34 — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43467)* status: 8 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 6 of 7 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43467)* status: 1. unresolved discussions.

**Review by RishabhGupta34 — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43467)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by deepak-harness — APPROVED** [review]
> Reviewed 6 of 7 files at r1, 1 of 1 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43467)* status: 1. unresolved discussions.

**Pull request merged**